src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,127 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int STANDARD_CAPACITY = 10;
+    private T[] arrayList;
+    private int size;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[STANDARD_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (hasArraySpace()) {
+            arrayList[size] = value;
+            size++;
+            return;
+        }
+        extendCapacity();
+        arrayList[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (!hasArraySpace()) {
+            extendCapacity();
+        }
+        if (index < 0 || index > arrayList.length || index > size + 1) {
+            throw new ArrayListIndexOutOfBoundsException("Your index does not suit rules");
```

Comment:
```suggestion
            throw new ArrayListIndexOutOfBoundsException("Wrong index.");
```
if to keep it simple

Reply:
done